hw/ip/spi_host/doc/_index.md

```diff
@@ -607,17 +824,100 @@ The Byte Select, or `bytesel`, unit is responsible for unpacking data from the F
+  {name:"sd[3]", wave:"xz.......22222222zz22334455",
+   data:["a[20]", "a[16]", "a[12]", "a[8]", "a[4]", "a[0]", "1", "1"]},
+  {node: ".A.......B.C.D.E.F.G.H.I.J.K"},
+  {node: ".........L.....M...N........O"}
+],
+  edge: ['A<->B Command 0xEB ("Fast Read Quad I/O")',  'B<->C MSB(addr)', 'D<->E LSB(addr)',
+         'G<->H addr[0]', 'H<->I addr[1]', 'I<->J addr[2]', 'J<->K addr[3]',
+         'L<->M Address', 'N<->O Data'],
+
+ foot: {text: "Addresses are transmitted MSB first, and data is returned in order of increasing peripheral byte address."}}
+{{< /wavejson >}}
+
+Note two important details:
+1. Each byte of data is returned in order of increasing address.
+2. All fields are transmitted most significant bit first, and for any multi-byte quantities (notably here the address), most significant bytes are transmitted first.
+In this regard, these flash devices are similar to Big-Endian CPU architectures.
+
+
+**TODO** (Complete programming reference outlined below)
+
+The first step in SPI_HOST we must first enable it.
+
+## Overview
 
 ## Basic Configuration
 
-## Issuing a Standard SPI command
+Next the SPI_HOST must be configured to match the needs of the SPI peripheral being targeted.
+
+```
+// Set CPOL to 0, CPHA to 0, and FULLCYC to 1
+
+// Leave CSAAT, CSNLEAD, CSNTRAIL, CSNIDLE can typically be left at
+// zero (default value).
+
+// Optional: set CLKDIV
+```
+
+## Loading the TX FIFO
+
+### Example 1: Quad Read with 4-Byte Addressing
+
+
+Larger SPI Flashes support both 4-byte and 3-byte addressing modes.
+However some smaller parts only support the 3-bytes addresses.
+
+**Note when using Little-Endian systems (such as Ibex)**: 32-bit integer addresses must be byte-swapped to Big-Endian byte-order before being loaded into {{< regref "TXDATA" >}}, because these SPI flash devices expect the MSB to be sent first.
+Assuming the SPI_HOST is compiled with `ByteOrder = 1`, sending the address without this byte swap will result in the MSB being sent last, and the address will be incorrectly received at the flash device.
+The `ByteOrder` parameter is meant to correct for byte-ordering issues that may be caused when using 32-bit instructions to copy bytes from *memory* to the SPI bus (i.e. the goal is to avoid shuffling of flash data).
+When copying 32-bit *CPU register* values, such as addresses, to {{< regref "TXDATA" >}} this byte-ordering correction (which is applied to properly order memory transfers) inverts the register address.
+
+### Example 2: Quad Read with 3-byte Addressing
+
+For 3-byte address commands, writing the 24-bit address to {{< regref "TXDATA" >}} as a 32-bit integer will *not* work.
+The address will be sent as a 4-byte value, and the entire command will be shifted incorrectly.
+There are two possible solutions to this problem:
+1. The instruction code can be packed into the MSB of the address, and written to {{< regref "TXDATA" >}}, to create a 4-byte header which can be loaded directly into {{< regref "TXDATA">}}
```

Comment:
would this not cause the Command to be transmitted in quad mode (i.e shifting the address out of position I don't think this is legal behavior) or the address to be transmitted in standard mode?
I think some waves would be great here to explain this?

Reply:
could the byte enables not be used to align the data correctly?

Reply:
This whole final section appears to be out of date.  (Though I personally claim it is not needed for D1).   I'm in the progress of updating it, but for the time being I'll just cut it out of this PR.

Reply:
This section will be moved to another PR.   (Simply because I don't think these details are needed for D1).   Normally I would preserve this comment, but I think the text will change so much that it might be best requesting you take a fresh look in the next round.